core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -993,6 +1000,15 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def handleApiVersionRequest(request: RequestChannel.Request) {
+    val responseHeader = new ResponseHeader(request.header.correlationId)
+    val responseBody = if (!authorize(request.session, Describe, Resource.ClusterResource))
+      ApiVersionResponse.fromError(Errors.CLUSTER_AUTHORIZATION_FAILED)
```

Comment:
It is usually treated as security flaw to be able to know what version is supported by server, as with that knowledge one can exploit security flaws associated with the version. Does cluster describe seem too restrictive? Trying to think what else can be done here. Do you have any suggestion?